### DISCUSSION TITLE
Potential changes to the default resource limits

### DISCUSSION THREAD

**Discussion by dooman87** [ImageMagick/ImageMagick#7091] [Development] [answered]
### ImageMagick version

7.1.1-27

### Operating system, version and so on

Linux Debian Bookworm

### Description

I'm using [Golang binding](https://github.com/gographics/imagick) for calling MagickWand and the code was working on version of IM 7.1.1-22. When I upgraded IM to 7.1.1-27 the `ReadImageBlob` call started to throw an error: 

```
EXCEPTION_ERROR: time limit exceeded ' @ error/cache.c/GetImagePixelCache/1703`
```

The code is pretty casual:

```
	mw := imagick.NewMagickWand()

	err = mw.ReadImageBlob(src.Data)
	if err != nil {
		return false, err
	}
```

Setting the limit manually is solving the problem:

```
	mw := imagick.NewMagickWand()

	err := mw.SetResourceLimit(imagick.RESOURCE_TIME, 90)
	if err != nil {
		return false, err
	}

	err = mw.ReadImageBlob(src.Data)
	if err != nil {
		return false, err
	}
```

However, I wonder if there were any changes in IM recently that could impact the default limits?

Please, note that `ReadImageBlob` in that case is taking less than a second, so the actuall execution time wouldn't be the problem, I think.

TIA!

**Comment by snibgo** [comment]
What is the text output from `magick -list policy`?

**Comment by dlemstra** [comment, marked as answer]
I just pushed some patches that should resolve this in the next release.

**Reply by dooman87** [reply]
Thanks legend! Just tested and everything is back to normal on the latest `main`!
